tests/test_prettytable.py

```diff
@@ -1912,6 +1912,59 @@ def test_jupyter_repr(self, row_prettytable: PrettyTable):
         assert row_prettytable._repr_html_() == row_prettytable.get_html_string()
 
 
+class TestMinTableWidth:
+    @pytest.mark.parametrize(
+        "loops, fields, desired_width, border, internal_border",
+        [
+            (15, ["Test table"], 20, True, False),
+            (16, ["Test table"], 21, True, False),
+            (18, ["Test table", "Test table 2"], 40, True, False),
+            (19, ["Test table", "Test table 2"], 41, True, False),
+            (21, ["Test table", "Test col 2", "Test col 3"], 50, True, False),
+            (22, ["Test table", "Test col 2", "Test col 3"], 51, True, False),
+            (19, ["Test table"], 20, False, False),
+            (20, ["Test table"], 21, False, False),
+            (25, ["Test table", "Test table 2"], 40, False, False),
+            (26, ["Test table", "Test table 2"], 41, False, False),
+            (25, ["Test table", "Test col 2", "Test col 3"], 50, False, False),
+            (26, ["Test table", "Test col 2", "Test col 3"], 51, False, False),
+            (18, ["Test table"], 20, False, True),
+            (19, ["Test table"], 21, False, True),
+            (23, ["Test table", "Test table 2"], 40, False, True),
+            (24, ["Test table", "Test table 2"], 41, False, True),
+            (22, ["Test table", "Test col 2", "Test col 3"], 50, False, True),
+            (23, ["Test table", "Test col 2", "Test col 3"], 51, False, True),
+        ],
+    )
+    def test_min_table_width(
+        self, loops, fields, desired_width, border, internal_border
+    ):
+        for col_width in range(loops):
+            x = prettytable.PrettyTable()
+            x.border = border
+            x.preserve_internal_border = internal_border
+            x.field_names = fields
+            x.add_row(["X" * col_width] + ["" for _ in range(len(fields) - 1)])
+            x.min_table_width = desired_width
+            t = x.get_string()
+            if border == False and internal_border == False:
+                assert [len(x) for x in t.split("\n")] == [desired_width, desired_width]
+            elif border == False and internal_border == True:
```

Comment:
To appease the linter:

```
tests/test_prettytable.py:1950:23: E712 comparison to False should be 'if cond is False:' or 'if not cond:'
tests/test_prettytable.py:1950:52: E712 comparison to False should be 'if cond is False:' or 'if not cond:'
tests/test_prettytable.py:1952:25: E712 comparison to False should be 'if cond is False:' or 'if not cond:'
tests/test_prettytable.py:1952:54: E712 comparison to True should be 'if cond is True:' or 'if cond:'
```

```suggestion
            if border is False and internal_border is False:
                assert [len(x) for x in t.split("\n")] == [desired_width, desired_width]
            elif border is False and internal_border is True:
```